endpoints/v2/__init__.py

```diff
@@ -127,8 +128,12 @@ def wrapped(namespace_name, repo_name, *args, **kwargs):
 
             repository = namespace_name + "/" + repo_name
             if allow_public:
-                repository_ref = registry_model.lookup_repository(namespace_name, repo_name)
-                if repository_ref is None or not repository_ref.is_public:
+                try:
+                    repository_ref = registry_model.lookup_repository(namespace_name, repo_name)
+                except RepositoryDoesNotExist:
+                    raise Unauthorized(repository=repository, scopes=scopes)
```

Comment:
I suspect this is not always correct to reply with Unauthorized in this case.

Reply:
Reverted back to previous code.